### ISSUE TITLE
Time Tracking Data Format

### ISSUE THREAD

**Issue by Goettcke** [Goettcke/TimeTracker#4]
Time tracking data should be saved to a file on disk, and this data should then be loaded into the program  when the program is started. 
The correct format has to be decided, and how to store the time data within the file for instance as ";" like in .csv or ", " or  " " separated. 
The final data format is pretty important, so I suggest a discussion about this

**Comment by PeterSR** [comment]
My vote is on a csv like format which allows users to easily open the tracking data in other csv-compatible programs.

Regarding what fields to use, I think it is important to know how the program is supposed to be used.

For instance, one could imagine running

```bash
./timetracker.py <project id> start
```

to start recording time and

```bash
./timetracker.py <project id> stop
```

to stop. In that case, the fields required would be: project id, event (start or stop), timestamp. 
From that we can compute the work time for that session.
Now to allow for other use cases and more advanced feature, we should probably plan ahead and include other fields or model it slightly different.

**Comment by Goettcke** [comment]
Yes I think that is great point. I would go for .csv aswell.  I just talked with Danni who is a future collaborator, and I told him my future Idea for the simple Gui. I think it should include a scrollable menu, where you pick the project to work on and simply click start. We need terminal activation aswell of course, cause when I attach to a tmux session, I would like the timer just to start automatically on that project, and when I close the session It should automatically stop the timer. 

One of the big problems I see with timer programs is that you have to manage the program. Imagine being able to do your workflow and the timer runs in the background without you managing it.

**Comment by Goettcke** [comment]
`      LOGINFO( errormsg1, strlen(errormsg1));` Sry just trying this coding thing

**Comment by PeterSR** [comment]
I agree that there should be a GUI as well as a terminal interface. In that case I suggest we create a sort of engine/core that any user interface can interface with (as to avoid code duplication for each interface). The core would then know how to read and write the data file, do timing etc. 

The automatic start and stop is the neat, but also hard part! In the case of tmux one could probably create a hook that runs when a session of a certain id is created and another when that session is closed. This is of course specific to tmux, so perhaps we should have a collection of hooks for different environments like bash, tmux, vim, emacs etc. that people then add themselves. 
Unless of course we can find a way to make it rather seamless without having to handle all possible environments people would use.
